src/modules/Hooks.ts

```diff
@@ -416,25 +416,30 @@ export class Hooks {
 	 */
 
 	// Overload: running HookDefaultHandler: expect HookDefaultHandler return type
-	protected async run<T extends HookName>(registrations: HookRegistration<T, HookDefaultHandler<T>>[], visit: Visit | undefined, args: HookArguments<T>): Promise<Awaited<ReturnType<HookDefaultHandler<T>>>[]>; // prettier-ignore
+	protected async run<T extends HookName>(registrations: HookRegistration<T, HookDefaultHandler<T>>[], visit: Visit | undefined, args: HookArguments<T>, rethrow: true): Promise<Awaited<ReturnType<HookDefaultHandler<T>>>[]>; // prettier-ignore
 	// Overload:  running user handler: expect no specific type
 	protected async run<T extends HookName>(registrations: HookRegistration<T>[], visit: Visit | undefined, args: HookArguments<T>): Promise<unknown[]>; // prettier-ignore
 	// Implementation
 	protected async run<T extends HookName, R extends HookRegistration<T>[]>(
 		registrations: R,
-		visit: Visit | undefined,
-		args: HookArguments<T>
+		visit: Visit | undefined = this.swup.visit,
+		args: HookArguments<T>,
+		rethrow: boolean = false
```

Comment:
Wouldn't `throw` be descriptive enough? I think I'd prefer that. Or is naming it `rethrow` following some kind of convention for that kind of behavior?

Reply:
I had throw, but it's a reserved word in JS :(

Reply:
ah! Makes sense 😁

Reply:
How about `throwErrors` then? I know you prefer one-word variables, but that I would understand immediately

Reply:
How do you feel about `throws`? with an s?

Reply:
I don't like it better than rethrow.

Reply:
Did you ask ChatGpt? 😉

Reply:
Alternatively, maybe we could call the variable `internal` or `builtin`. And then throw errors if `internal` is true.

Reply:
I did, and it said `rethrow` is the common word for that behavior. [Google seems to agree rethrowing is a thing](https://www.google.com/search?q=rethrow+error&oq=rethrow+error).

Reply:
I frankly don't mind much, it's an internal helper function with no public API, so let's go with `throwError` if `rethrow` doesn't make sense for you.

Reply:
Thanks for googling that for me 🤦😅 ...lets go with rethrow.